x/multisig/keeper/msg_server.go

```diff
@@ -60,11 +60,15 @@ func (s msgServer) SubmitPubKey(c context.Context, req *types.SubmitPubKeyReques
 		return nil, fmt.Errorf("sender %s is not a registered proxy", req.Sender.String())
 	}
 
-	err := keygenSession.AddKey(ctx.BlockHeight(), participant, req.PubKey)
+	completed, err := keygenSession.AddKey(ctx.BlockHeight(), participant, req.PubKey)
```

Comment:
why not use `GetState` to check keygen completeness? so add key don't need to return a bool

Reply:
We only want to update the expiry once, when the keygen state goes from pending to completed. Validators can still submit after completion due to grace period. So, I'd have to check that the state changed from pending to completed instead which bleeds more lower level logic here.